tensorflow/c/kernels.cc

```diff
@@ -279,4 +279,35 @@ TF_Tensor* TF_AllocateOutput(TF_OpKernelContext* context, int index,
     return nullptr;
   }
   return tf_tensor;
-}
\ No newline at end of file
+}
+
+TF_Tensor* TF_AllocateTemp(TF_OpKernelContext* context, TF_DataType dtype, 
+                           int64_t* dims, int num_dims, 
+                           TF_AllocatorAttributes attributes,
+                           TF_Status* status) {
+  auto* cc_ctx = reinterpret_cast<::tensorflow::OpKernelContext*>(context);
+  TF_SetStatus(status, TF_OK, ""); 
+  tensorflow::TensorShape shape;
+  for (int i = 0; i < num_dims; ++i) {
```

Comment:
Instead of copying over the shape, can we instead use this constructor instead? https://github.com/tensorflow/tensorflow/blob/7a4b89dbaf78503882dfeefa634c5bbff303d563/tensorflow/core/framework/tensor_shape.h#L166-L168

The advantage is that this function knows the rank of the shape ahead of time, so it knows which internal representation to use for its union structure. 

Here's an example: https://github.com/tensorflow/tensorflow/blob/7a4b89dbaf78503882dfeefa634c5bbff303d563/tensorflow/c/kernels.cc#L267-L268